include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1031,8 +1031,8 @@ __pattern_remove_if(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec,
     //TODO: To optimize copy back depending on Iterator, i.e. set_final_data for host iterator/pointer
     // __pattern_copy_if above may be async due to there is implicit synchronization on sycl::buffer and the accessors
 
-    //An explicit wait isn't required here because we have implicit synchronization on sycl::buffer destructor.
-    return __pattern_walk2</*_IsSync=*/std::false_type>(
+    //A buffer is constructed from a range, the destructor does not need to block.
```

Comment:
I don't think any comment is required here anymore.  If we want some comment here, perhaps: 

//__pattern_walk2 is a blocking call

Similar comment for the repetitions of this later.

Reply:
I think to keep such comment is very important , because the situation (with misunderstanding what is going on)  might be repeated again by looking at the someone? not me, you or Sergey.. And that person also may decide that __pattern_walk2 may be called in async way "because" we have sycl::buffer above on the stack.

Reply:
Fair point, perhaps we just need to expand it a little more then:
```
// The temporary buffer is constructed from a range, therefore it's destructor will not block, therefore 
// we must call __pattern_walk2 in a way which provides blocking synchronization for this pattern.
```